proto/sdv/databroker/v1/types.proto

```diff
@@ -144,56 +145,54 @@ message Datapoint {
 }
 
 message Metadata {
-  int32 id               = 1;
-  EntryType entry_type   = 2;
-  string name            = 4;
-  DataType data_type     = 5;
-  ChangeType change_type = 6;  // CONTINUOUS or STATIC or ON_CHANGE
-  string description     = 7;
-
-  ValueRestriction value_restriction = 10;
-  // int32             min_update_hz       = 10; // Only for CONTINUOUS
-  // int32             max_update_hz       = 11; // Only for CONTINUOUS
+  int32 id                             = 1;
+  EntryType entry_type                 = 2;
+  string name                          = 4;
+  DataType data_type                   = 5;
+  ChangeType change_type               = 6;  // CONTINUOUS or STATIC or ON_CHANGE
+  string description                   = 7;
+
+  // Value restrictions checked/enforced by Databroker.
+  ValueRestrictions value_restrictions = 10;
+}
+
+// Restriction fields are checked/enforced by Databroker.
+// They can be either values specified in VSS, or if not present implicit limits for the datatype.
+message ValueRestrictions {
+  Allowed allowed      = 1;
+  ValueRestriction min = 2;
+  ValueRestriction max = 3;
+}
```

Comment:
I don't see why we need to create an additional message to contain the value restrictions. All the other metadata is at the "root" level which IMO makes it easier to work with.

I think this should be reverted to what was already there, i.e.

```
message Metadata {
  ..
  Allowed allowed = 10;
}
```

and support for mix/max can be added along these lines:
```rust
message Metadata {
  ..
  Allowed allowed      = 10;
  ValueRestriction min = 11;
  ValueRestriction max = 12;
}
```

Reply:
First thing I tried was your suggestion, but I decided to wrap all in same message since the three messages are somehow related with each other to restrict user's update values and also the way it looks now the conversion call is more (imo) cleaner and easy to understand with just one message field (see-> https://github.com/eclipse-kuksa/kuksa-databroker/pull/96/files#diff-d482c91dd456ce51fbebd32a722dab7e339bfda360b1ab807c296ba95ca18179R383)

Reply:
I don't really see how putting these inside a `ValueRestriction` message makes it cleaner / easier to understand.

It seems to be the only field that requires the whole metadata struct in order to be constructed. And to construct it, first `metadata.allowed`, `metadata.min` and `metadata.max` are checked individually inside `value_restriction_from(...)` and if one of them is set a `ValueRestriction` is created:

```rust
fn value_restriction_from(metadata: &broker::Metadata) -> Option<proto::ValueRestrictions> {
  ...
  if min_value.is_some() | max_value.is_some() | allowed.is_some() {
     return Some(proto::ValueRestrictions { allowed: ..., min: ..., max: ...})
  }
  ...
}
```

How is that cleaner / easier to understand than how all the other metadata is converted, i.e. something like this:
```rust
proto::Metadata {
  allowed_values: metadata.allowed.into(),
  min: metadata.min.into(),
  max: metadata.max.into(),

}
```

Reply:
Done